### PR TITLE
Moves metadata to its own class accessed through Items

### PR DESCRIPTION
**Pull request by digitaldan**
also adds getConfigurationValue

Signed-off-by: Dan Cunningham <dan@digitaldan.com>

### CODE REVIEW

File: metadata/metadata.js [hunk 1]
```diff
@@ -7,70 +7,113 @@
...
- */
-let upsertValue = function(itemName, namespace, value) {
-    let existing = getValue(itemName, namespace);
+    /**
+    * Adds (inserts) a metadata value.
+    * 
+    * @param {String} namespace the name of the namespace
+    * @param {String} value the value to insert or update
+    */
+    addValue(namespace, value) {
```
Comment (marcinczeczko) [inline metadata/metadata.js:50]: I think it's worth to take a full potential of metadata and let the `addValue` method to set also metadata configuration values. currently this method sets it to empty. I was thinking about splitting it into two methods as for the reading configuration values, but apparently the `Metadata` object doesn't have any methods - it's possible only via constructor

File: metadata/metadata.js [hunk 2]
```diff
@@ -7,70 +7,113 @@
...
-    } else {
-        updateValue(itemName, namespace, value);
-        return false;
+    /**
+    * Updates metadata values for this item.
+    * @param {String} namespace The namespace for the metadata to update
+    * @param {String} value the value to update the metadata to
+    * @returns {String} the updated value
+    */
+    updateValue(namespace, value) {
```
Comment (marcinczeczko) [inline metadata/metadata.js:61]: If `addValue` would support metadata.configuration, then the update method should do the same

File: metadata/metadata.js [hunk 3]
```diff
@@ -7,70 +7,113 @@
...
+        let metadata = createMetadata(this.itemName, namespace, value);
+        let result = MetadataRegistry.update(metadata);
+        return result ? result.value : null;
+    }
+
+    /**
+     * Updates metadata values for this item.
+     * @param {Map} namespaceToValues A map of namespaces to values to update
+     */
+    updateValues(namespaceToValues) {
```
Comment (marcinczeczko) [inline metadata/metadata.js:71]: Here the `namespaceToValues` map should support configuration too. But it's worth to document how to prepare that structure. Wdyt ?

File: metadata/metadata.js [hunk 4]
```diff
@@ -7,70 +7,113 @@
...
+    }
+
+    /**
+     * Adds (inserts) or updates a metadata value.
+     * 
+     * @param {String} namespace the name of the namespace
+     * @param {String} value the value to insert or update
+     * @returns {Boolean} true if the value was added, false if it was updated
+     */
+    upsertValue(namespace, value) {
```
Comment (marcinczeczko) [inline metadata/metadata.js:84]: And of course `upsertValue` should handle metadata.configuration too

**Comment by digitaldan** [comment]
Thanks @marcinczeczko for the feedback, I'll take a look at supporting updating the configuration options fully if we are ok with this approach.

**Pull request closed without merging**